datasync/chaindatafetcher/kas/repository.go

```diff
@@ -0,0 +1,55 @@
+package kas
+
+import (
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jinzhu/gorm"
+	"github.com/klaytn/klaytn/log"
+	"time"
+)
+
+const (
+	maxTransactionCount         = int64(1000000)
+	maxTransactionLogCount      = int64(100000)
+	maxInternalTransactionCount = int64(10000)
+
+	maxPlaceholders = 65535
+
+	placeholdersPerTxItem = 13
+
+	maxDBRetryCount = 20
+	DBRetryInterval = 1 * time.Second
+)
+
+var logger = log.NewModuleLogger(log.ChainDataFetcher)
+
+type repository struct {
+	db *gorm.DB
+}
+
+func getEndpoint(user, password, host, port, name string) string {
+	return user + ":" + password + "@tcp(" + host + ":" + port + ")/" + name + "?parseTime=True&charset=utf8mb4"
+}
+
+func NewRepository(user, password, host, port, name string) (*repository, error) {
+	endpoint := getEndpoint(user, password, host, port, name)
+	var (
+		db  *gorm.DB
+		err error
+	)
+	for i := 0; i < maxDBRetryCount; i++ {
+		db, err = gorm.Open("mysql", endpoint)
+		if err != nil {
+			logger.Info("Retrying to connect DB", "endpoint", endpoint, "err", err)
```

Comment:
Shouldn't this be `Warn`?

Reply:
Right. Chainged it 9fa1c7acb42f8f23587d26b76a1976565d98b074.